spectator/sidecarconfig.py

```diff
@@ -0,0 +1,53 @@
+from typing import Dict, Any
+
+from spectator.commontags import common_tags
+
+
+class SidecarConfig:
+    """Sidecar Configuration Class
+
+    To provide a custom configuration for Spectator-py, pass a dictionary to the constructor which
+    returns values for the keys listed below. Omitting keys will cause this class to return the
+    default values.
+
+    {
+      "sidecar.common-tags": {"key": "value"},
+      "sidecar.debug-logging": True,
+      "sidecar.output-location": "location"
+    }
+
+    Then provide the custom instance to a new instance of the Registry, rather than relying on
+    the GlobalRegistry, which has a default configuration.
+    """
+
+    def __init__(self, config: Dict[str, Any] = None) -> None:
+        """Optionally initialize with a dictionary which returns values that override the
+        defaults."""
+        if config is None:
+            self._config = {}
+        else:
+            self._config = config
+
+    def common_tags(self) -> Dict[str, str]:
+        """Common infrastructure tags from the Netflix environment variables."""
+        return self._config.get("sidecar.common-tags", common_tags())
+
+    def debug_logging(self) -> bool:
```

Comment:
Instead of having a separate config option for this, why not have the writer call `logger.isEnabledFor` and cache the result? For example:

https://docs.python.org/3/howto/logging.html#optimization

The main benefit of this would be that from a user perspective all they have to do is adjust the log level. So there would be a single authoritative setting rather than two flags that have to get set correctly to get the logging.

Reply:
Adding a caching scheme that checks the level on the first write - this maintains the overall performance and provides an easy hook to increase logging output.